oh_staff_ui/views_utils.py

```diff
@@ -466,3 +470,105 @@ def run_process_file_command(
     # testing confirms this, though it does close after 30 seconds or so.
     # To be safe, explicitly close this one when done.
     connection.close()
+
+
+def get_record_oai(ark: str = None, req_url: str = None) -> str:
+    return get_listrecords_oai("GetRecord", ark, req_url)
+
+
+def get_listrecords_oai(verb: str, ark: str = None, req_url: str = None) -> str:
+
+    pi_set = ProjectItem.objects.filter(status__status__iexact="completed")
+
+    if ark:
+        pi_set = pi_set.filter(ark=ark)
+
+    verb_element = etree.Element(verb)
+
+    for pi in pi_set:
+        verb_element.append(add_oai_envelope_to_mods(OralHistoryMods(pi)))
+
+    return wrap_oai_content(verb_element, verb, ark, req_url)
+
+
+def wrap_oai_content(
+    xml_element: etree.Element, verb: str, ark: str, req_url: str
+) -> str:
+
+    oai_tree = get_oai_envelope()
+    oai_tree.append(get_response_date_element())
+    oai_tree.append(get_request_element(verb, ark, req_url))
+    oai_tree.append(xml_element)
+
+    return etree.tostring(oai_tree)
+
+
+def get_oai_envelope() -> etree.Element:
+    oai_envelope = """
+            <OAI-PMH xmlns="http://www.openarchives.org/OAI/2.0/" 
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+                xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/ 
+                http://www.openarchives.org/OAI/2.0/OAI-PMH.xsd" />
+            """
+    oai_tree = etree.fromstring(oai_envelope)
+
+    return oai_tree
+
+
+def get_response_date_element() -> etree.Element:
+
+    date_el = etree.Element("responseDate")
+    date_el.text = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    return date_el
+
+
+def get_request_element(
+    verb: str, ark: str = None, req_url: str = None
+) -> etree.Element:
+
+    req = f"""<request metadataPrefix="mods">{req_url}</request>"""
+
+    e_req = etree.fromstring(req)
+    e_req.set("verb", verb)
+
+    if ark:
+        e_req.set("identifier", ark)
+
+    return e_req
+
+
+def get_bad_arg_error_xml(verb: str, req_url: str = None) -> str:
```

Comment:
Please document this method.  I see it adds a bit of XML, but there's no context here; I have to go back to the `oai` view to see it gets called when `GetRecord` verb is used with no ARK.  More information would be helpful, like why do this (returning HTTP 200 with this error information in the XML) rather than returning an HTTP 400 or similar.